lib/iris/analysis/_regrid.py

```diff
@@ -452,7 +455,13 @@ def regrid_reference_surface(src_surface_coord, surface_dims,
                 warnings.warn(msg)
         return result
 
-    def _check_units(self, coord):
+    @staticmethod
+    def _check_units(coord):
```

Comment:
i'd rather this wasn't changed to a static method, I think.

If we need to reuse the functionality, we should either factor it out to share, or reproduce a similar in experimental in the interim